docs/hazmat/backends/interfaces.rst

```diff
@@ -509,3 +509,12 @@ A specific ``backend`` may provide one or more of these interfaces.
         :param bytes data: DER formatted certificate data.
 
         :returns: An instance of :class:`~cryptography.x509.Certificate`.
+
+    .. method:: load_pem_x509_csr(data)
+
+        .. versionadded:: 0.9
+
+        :param bytes data: PEM formatted certificate request data.
```

Comment:
PEM is an abbrevaitation that is more well known than the expanded form
(and also the expanded form is basically meaningless), I don't think that's
true of CSR. That said, it /is/ a very long name.

On Sun, Mar 15, 2015 at 5:10 PM, Paul Kehrer <notifications@github.com>
wrote:

> In docs/hazmat/backends/interfaces.rst
> <https://github.com/pyca/cryptography/pull/1702#discussion_r26452379>:
>
> > @@ -509,3 +509,12 @@ A specific ``backend`` may provide one or more of these interfaces.
> >          :param bytes data: DER formatted certificate data.
> >
> >          :returns: An instance of :class:`~cryptography.x509.Certificate`.
> > +
> > +    .. method:: load_pem_x509_csr(data)
> > +
> > +        .. versionadded:: 0.9
> > +
> > +        :param bytes data: PEM formatted certificate request data.
>
> I discussed this with glyph a bit on IRC and it was his contention that
> load_pem_x509_csr is better than load_pem_x509_certificate_signing_request
> since pem is already an abbreviation. I am leaning towards leaving it as
> csr here but could be persuaded otherwise.
> load_pem_x509_certificate_signing_request is a pretty long name though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1702/files#r26452379>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084